File: frontend/src/types/attendance.ts
export interface Attendance {
  id: string;
  facility_id: string;
  staff_id: string;
  staff_name: string;
  check_in: string;
  check_out: string | null;
  notes: string | null;
  recorded_by_user_id: string;
  created_at: string;
}

export interface AttendanceCreate {
  facility_id: string;
  staff_id: string;
  check_in: string;
  notes?: string;
}

export interface AttendanceCheckOut {
  check_out: string;
  notes?: string;
}

export interface AttendanceReport {
  staff_id: string;
  staff_name: string;
  total_hours: number | null;
  check_ins: number;
  check_outs: number;
  incomplete_sessions: number;
}

File: backend/app/schemas/attendance.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime
from typing import Optional


class AttendanceCreate(BaseModel):
    facility_id: UUID
    staff_id: UUID
    check_in: datetime
    notes: Optional[str] = None


class AttendanceCheckOut(BaseModel):
    check_out: datetime
    notes: Optional[str] = None


class AttendanceResponse(BaseModel):
    id: UUID
    facility_id: UUID
    staff_id: UUID
    staff_name: str  # Denormalizado para facilitar listado
    check_in: datetime
    check_out: Optional[datetime]
    notes: Optional[str]
    recorded_by_user_id: UUID
    created_at: datetime

    class Config:
        from_attributes = True


class AttendanceReportResponse(BaseModel):
    staff_id: UUID
    staff_name: str
    total_hours: Optional[float]
    check_ins: int
    check_outs: int
    incomplete_sessions: int
